cowprotocol/cow-amm
# Add deploy scripts

Adds a few deployment scripts:
- Two to deploy each contract one by one (Uniswap oracle and constant product)
- One to deploy both of them at the same time.

This is done because I expect in the future to only use the single deployments every time we introduce a new contract on an old chain, but we still want a way to deploy everything easily on a new chain.

Note that deployments are **not** deterministic. This is an intentional choice because the import structure we have can change and this would also change the deterministic address (installed libs update the remappings automatically, which are part of the contract metadata even if they are unused). This could be worked around but has been deprioritized for a speedier deployment.

I also added a few test to make sure the scripts don't revert.

Another note: I didn't want the `DeployAllContracts` script to inherit the other scripts, I'd have rather had an instance "deployed" inside the script and then call the deployment function. This didn't work because it caused the tests to mysteriously revert. I didn't investigate this too much but I suspect this is a bug in Foundry.

### How to test

There are a few dedicated tests. Soon there's also going to be a PR where you can check the result of running the deploy-all script.
